src/main/java/mate/academy/AsyncRequestProcessor.java

```diff
@@ -1,16 +1,48 @@
 package mate.academy;
 
+import java.util.Map;
 import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.Executor;
+import mate.academy.exception.NoUserInCacheException;
+import mate.academy.exception.RequestProcessingException;
 
 public class AsyncRequestProcessor {
+    private static Map<String, UserData> cache = new ConcurrentHashMap<>();
     private final Executor executor;
 
     public AsyncRequestProcessor(Executor executor) {
         this.executor = executor;
     }
 
     public CompletableFuture<UserData> processRequest(String userId) {
-        return null;
+        return CompletableFuture.supplyAsync(
+                () -> findUserInCache(userId), executor).exceptionally(ex -> {
+                    try {
+                        UserData userData = emulateDataBaseAccess(userId);
+                        CompletableFuture.runAsync(() -> cacheFetchedData(userId, userData));
```

Comment:
The `runAsync` method is not waiting for the cache update operation to complete, which may lead to inconsistent state if the `cacheFetchedData` method is not atomic or if the cache update is critical to subsequent operations. Consider using `thenAcceptAsync` after the `supplyAsync` method to ensure proper ordering.